include/af/image.h

```diff
@@ -1429,6 +1463,48 @@ extern "C" {
                           const unsigned sobel_window, const bool is_fast);
 #endif
 
+#if AF_API_VERSION >= 36
+    /**
+       C Interface for gradient anisotropic(non-linear diffusion) smoothing
+
+       \param[out] out is an af_array containing anisotropically smoothed image pixel values
+       \param[in] in is the input image, expects non-integral (float/double) typed af_array
+       \param[in] timestep is the time step used in solving the diffusion equation.
+       \param[in] conductance parameter controls the sensitivity of conductance in diffusion equation.
+       \param[in] iterations is the number of times the diffusion step is performed.
+       \param[in] fftype indicates whether quadratic or exponential flux function is used by algorithm.
+       \return \ref AF_SUCCESS if the moment calculation is successful,
+       otherwise an appropriate error code is returned.
+
+       \ingroup image_func_gradient_anisotropic_diffusion
+    */
+    AFAPI af_err af_gradient_anisotropic_diffusion(af_array* out, const af_array in,
```

Comment:
Don't you think it would be better to combine the to anisotropic functions into one? They seem to have the same arguments.

Reply:
I wasn't sure how many other kinds of anisotropic diffusion are possible, so I couldn't decide if to use bool or an enum. Therefore, just added both separately. Lets, just merge them and deal with such scenario if and when it arises.